qos-monitor/src/main/java/eu/arrowhead/core/qos/service/ping/monitor/PingEventCollectorTask.java

```diff
@@ -0,0 +1,153 @@
+	private static final String REPLACEING_MEASURMENT_EVENT = " - measurment , duplicate event. Overwriting : ";
+
+	private final long clearingInterval = 1000 * 60 * 10;
+
+	private boolean interrupted = false;
+	private long lastBufferCleanAt;
+
+	@Resource(name = QosMonitorConstants.EVENT_QUEUE)
+	private LinkedBlockingQueue<EventDTO> eventQueue;
+
+	@Resource(name = QosMonitorConstants.EVENT_BUFFER)
+	private ConcurrentHashMap<UUID, PingEventBufferElement> eventBuffer;
+
+	@Autowired
+	private PingEventBufferCleaner bufferCleaner;
+
+	private final Logger logger = LogManager.getLogger(PingEventCollectorTask.class);
+
+	//=================================================================================================
+	// methods
+
+	//-------------------------------------------------------------------------------------------------	
+	@Override
+	public void run() {
+		logger.debug("PingEventCollectorTask run started...");
+
+		interrupted = Thread.currentThread().isInterrupted();
+		clearBuffer();
+
+		while (!interrupted) {
+			logger.debug("PingEventCollectorTask run loop started...");
+
+			try {
+				putEventToBuffer(eventQueue.take());
+
+				if (lastBufferCleanAt + clearingInterval < System.currentTimeMillis()) {
+
+					lastBufferCleanAt = System.currentTimeMillis();
+					clearBuffer();
+				}
+
+			} catch (final InterruptedException ex) {
+
+				logger.debug("PingEventCollectorTask run intrrupted");
+				interrupted = false;
+
+			}catch (final Throwable ex) {
+
+				logger.debug(ex.getMessage());
+			}
+		}
+	}
+
+	//=================================================================================================
+	// assistant methods
+
+	//-------------------------------------------------------------------------------------------------
+	private void clearBuffer() {
+		logger.debug("clearBuffer started...");
+
+		bufferCleaner.clearBuffer();
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void addEventToBufferElement (final PingEventBufferElement element, final int position, final MeasurementMonitoringEvent event) {
+		logger.debug("addEventToBufferElement started...");
+
+		if(element.getEventlist()[position] != null) {
+			logger.warn(element.getId() + REPLACEING_MEASURMENT_EVENT + element.getEventlist()[position].toString());
+		}
+
+		element.addEvent(position, event);
+
+	}
+
+	//-------------------------------------------------------------------------------------------------
+	private void putEventToBuffer(final EventDTO event) {
+		logger.debug("putEventToBuffer started...");
+
+		final UUID id = UUID.fromString(event.getMetaData().get(QosMonitorConstants.PROCESS_ID_KEY));
+
+		PingEventBufferElement element = eventBuffer.get(id);
+		if (element == null) {
+			element = new PingEventBufferElement(id);
+		}
+
+		try {
+			final QosMonitorEventType eventType = QosMonitorEventType.valueOf(event.getEventType());
```

Comment:
this one throws an IllegalArgumentException if event.getEventType() is no good. You should catch that one too with the ArrowheadException.

Reply:
fixed